src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -7,297 +7,405 @@
-                    factorsByVarConfig.computeIfAbsent(varConfig, k -> new SensitivityFactorGroup())
-                            .getFactors().add(injectionFactor);
+    protected void fillRhsContingency(final LfNetwork lfNetwork, final EquationSystem equationSystem, final Map<String, ComputedContingencyElement> contingencyElements, final Matrix rhs) {
+        for (ComputedContingencyElement element : contingencyElements.values()) {
+            if (element.getElement().getType().equals(ContingencyElementType.BRANCH)) {
+                LfBranch lfBranch = lfNetwork.getBranchById(element.getElement().getId());
+                if (lfBranch.getBus1() == null || lfBranch.getBus2() == null) {
+                    continue;
                 }
-            } else if (factor instanceof BranchFlowPerPSTAngle) {
-                BranchFlowPerPSTAngle pstAngleFactor = (BranchFlowPerPSTAngle) factor;
-                String phaseTapChangerHolderId = pstAngleFactor.getVariable().getPhaseTapChangerHolderId();
-                TwoWindingsTransformer twt = network.getTwoWindingsTransformer(phaseTapChangerHolderId);
-                if (twt == null) {
-                    throw new PowsyblException("Phase shifter '" + phaseTapChangerHolderId + "' not found");
+                LfBus bus1 = lfBranch.getBus1();
+                LfBus bus2 = lfBranch.getBus2();
+                if (bus1.isSlack()) {
+                    Equation p = equationSystem.getEquation(bus2.getNum(), EquationType.BUS_P).orElseThrow(IllegalStateException::new);
+                    rhs.set(p.getColumn(), element.getGlobalIndex(), -1 / PerUnit.SB);
+                } else if (bus2.isSlack()) {
+                    Equation p = equationSystem.getEquation(bus1.getNum(), EquationType.BUS_P).orElseThrow(IllegalStateException::new);
+                    rhs.set(p.getColumn(), element.getGlobalIndex(), 1 / PerUnit.SB);
+                } else {
+                    Equation p1 = equationSystem.getEquation(bus1.getNum(), EquationType.BUS_P).orElseThrow(IllegalStateException::new);
+                    Equation p2 = equationSystem.getEquation(bus2.getNum(), EquationType.BUS_P).orElseThrow(IllegalStateException::new);
+                    rhs.set(p1.getColumn(), element.getGlobalIndex(), 1 / PerUnit.SB);
+                    rhs.set(p2.getColumn(), element.getGlobalIndex(), -1 / PerUnit.SB);
                 }
-                SensitivityVariableConfiguration varConfig = new SensitivityVariableConfiguration(Collections.emptyMap(), Collections.singleton(phaseTapChangerHolderId));
-                factorsByVarConfig.computeIfAbsent(varConfig, k -> new SensitivityFactorGroup())
-                        .getFactors().add(pstAngleFactor);
             } else {
-                throw new UnsupportedOperationException("Factor type '" + factor.getClass().getSimpleName() + "' not yet supported");
+                throw new UnsupportedOperationException("Only contingency on a branch is yet supported");
```

Comment:
You check contingencies in the very beginning of your code. Do we really need to keep that exception ?